### ISSUE TITLE
[Feature Request] Show `zola serve` errors in web browser.

### ISSUE THREAD

**Issue by grantshandy** [Keats/gutenberg#2303]
I think it would be very useful to also serve a basic page with the build error instead of just print it to the CLI and stop the server. I think it would speed up zola users' workflow, and would not be *that* difficult to implement a basic error template.

I'd be interested in adding this feature, let me know if this is something that is wanted.

**Comment by Keats** [comment]
It would be nice to have yes

**Comment by stellarpower** [comment]
Using version 0.19.1, I get messages like these if zola starts successfully and I then edit the site. However, I want to present the site to users who will be creating the content but aren't developers. I therefore have Zola in a container and set to run repeatedly on error. If it restarted for whatever reason and the site can;t be processed, then my users won't be able to see the error messages, and then it'll also back off and loop over and over without starting up.

Is it possible to have `zola serve` continue to run even if the site can't be processed on startup, and then present the errors to the browser unconditionally?

Thanks

**Comment by Keats** [comment]
> Is it possible to have zola serve continue to run even if the site can't be processed on startup, and then present the errors to the browser unconditionally?

We can't really I think. You might have invalid templates so we can't display anything

**Comment by stellarpower** [comment]
Wouldn't it be possible to output the same error message as goes to the CLI to a generic webpage though? It seems this happens if I start the server and then break my site, but not if it's broken on startup. Otherwise I would either need to wrap the process in another webserver to print stdout/err or log in and go and check the container logs if one of the end users has a problem.

**Comment by stellarpower** [comment]
![image](https://github.com/user-attachments/assets/80c8fb97-4da2-412f-b5a1-656f59d59aee)

This is what I get sporadically. Not sure why sometimes it does show, and others it doesn't. If I load any page in a different tab, I get back a 404. I guess is it perhaps only if it hot-reloads that I see the message, over the websocket?

**Comment by EvanCarroll** [comment]
@stellarpower That's because you get that error only after you load a page, and you've created the failure mode. That is to say the error wasn't going across websockets. You'd create a build error, when you click a link you'd see the Zola Build error, and then when you resolved that problem you'd have to hit refresh. This is all solved in #3048
